### PR TITLE
Add sphinx document generation with tox

### PR DESCRIPTION
**Pull request by vmatyusGitHub**
## PR summary

Relates to: sdks/doc-generation

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [ ] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?

No python doc is available.

## What is the new behavior?

This PR contains changes that needed for python document generation.
To generate the document you need to call from the project root: `python3 -m tox -e docs`
The generated doc is published on GH pages, that is available from here: https://ibm.github.io/cloudant-python-sdk/
The doc publish process is not merged yet. Here you can also review it: https://github.com/IBM/cloudant-python-sdk/pull/45

This PR contains changes that needed for python doc generation.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: source/conf.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Configuration file for the Sphinx documentation builder.
```
Comment (bessbd) [inline source/conf.py:1]: I can't tell if this file needs copyright / license headers.

**Review by bessbd — APPROVED** [review]
Besides that one comment I've left, the change looks good to me.

File: source/conf.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import sys
+sys.path.insert(0, os.path.abspath('..'))
+print(os.path.abspath('..'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Cloudant Python SDK'
+copyright = '2021, (C) Copyright IBM Corp'
+author = 'cloudant-team'
```
Comment (mojito317) [inline source/conf.py:23]:
```suggestion
author = '@IBM/cloudant-sdks'
```

**Review by mojito317 — APPROVED** [review]
LGTM, only one suggestion.

File: source/conf.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import os
+import sys
+sys.path.insert(0, os.path.abspath('..'))
+print(os.path.abspath('..'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Cloudant Python SDK'
+copyright = '2021, (C) Copyright IBM Corp'
```
Comment (ricellis) [inline source/conf.py:22]: Is this "year, statement" a specific format required by the tool?
If not we should replace with:
```suggestion
copyright = '© Copyright IBM Corp. 2021.'
```

File: source/conf.py (hunk 1 above)
Comment (vmatyusGitHub) [reply source/conf.py:1]: This file is a setting file and adds no creative content, so I suppose it is not mandatory. - But I am not a copyright expert.
However I found a shorter form of the licence and I can add this to this file: http://www.apache.org/foundation/license-faq.html#Apply-My-Software
WDYT?

File: source/conf.py (hunk 2 above)
Comment (vmatyusGitHub) [reply source/conf.py]: Brilliant suggestion to use the GitHub group! Thanks!

File: source/conf.py (hunk 1 above)
Comment (vmatyusGitHub) [reply source/conf.py:4]: Changed in 9e559f7

File: source/conf.py (hunk 3 above)
Comment (vmatyusGitHub) [reply source/conf.py]: The sphinx itself embraces the owner with copyright sign and closing dot.
But I could change the format! Here is the change: 9e559f7

**Comment by vmatyusGitHub** [comment]
Thank you for the review @bessbd and @ricellis . I am about to merge this.

**Pull request merged**